packages/core/core/src/requests/PathRequest.js

```diff
@@ -1,32 +1,67 @@
-// @flow
-
-import type {AssetRequestDesc, Dependency, ParcelOptions} from './types';
+// @flow strict-local
 import type {Diagnostic} from '@parcel/diagnostic';
-import type ParcelConfig from './ParcelConfig';
+import type ParcelConfig from '../ParcelConfig';
+import type {StaticRunOpts} from '../RequestTracker';
+import type {AssetGroup, Dependency, ParcelOptions} from '../types';
 
-import {PluginLogger} from '@parcel/logger';
-import {relatifyPath} from '@parcel/utils';
 import ThrowableDiagnostic, {errorToDiagnostic} from '@parcel/diagnostic';
+import {PluginLogger} from '@parcel/logger';
+import {escapeMarkdown, relatifyPath} from '@parcel/utils';
 import path from 'path';
 import URL from 'url';
+import {report} from '../ReporterRunner';
+import PublicDependency from '../public/Dependency';
+import PluginOptions from '../public/PluginOptions';
+
+export type PathRequestInput = {|dependency: Dependency, config: ParcelConfig|};
+export type PathRequestResult = AssetGroup | null | void;
+
+export type PathRequest = {|
+  id: string,
+  +type: 'path_request',
+  run: RunOpts => Promise<PathRequestResult>,
+  input: PathRequestInput,
+|};
+
+type RunOpts = {|
+  input: PathRequestInput,
+  ...StaticRunOpts,
+|};
 
-import {report} from './ReporterRunner';
-import PublicDependency from './public/Dependency';
-import PluginOptions from './public/PluginOptions';
+const type = 'path_request';
 
-import {escapeMarkdown} from '@parcel/utils';
+export default function createPathRequest(input: PathRequestInput) {
+  return {
+    id: input.dependency.id,
+    type,
+    run,
+    input,
+  };
+}
+async function run({input, api, options}: RunOpts) {
+  let {config, dependency} = input;
+  let resolverRunner = new ResolverRunner({options, config});
+  let assetGroup = await resolverRunner.resolve(dependency);
+
+  // ? Should this happen if asset is deferred?
+  if (assetGroup != null) {
+    api.invalidateOnFileDelete(assetGroup.filePath);
+  }
+
```

Comment:
I think there are probably quite a few other cases we are missing here for when resolution could change. e.g. if you deleted/changed the package.json entry, added a node_modules directory closer to the source file, added/deleted an index.js file, etc. We should make sure we cover all of those at some point. Perhaps we need to pass the invalidation API to resolver plugins as well, or have them return some list of invalidations. I think it may be plugin specific in some cases.

Reply:
Do you think we pass the invalidation api directly or abstract it somewhat like we do with `includedFiles` in assets and configs?

Reply:
Not sure. IMO it would be nicer to have it return something rather than an imperative method, but not sure how it would look exactly. Let's get this in an issue/height task so we don't forget.